### ISSUE TITLE
Evaluation and training accuracy do not match

### ISSUE THREAD

**Issue by zliubabalala** [keras-team/keras#9694]
I found some really weird stuff in the Keras 2.1.5. I have tested the pre-train MobileNet on the dogs and cats data set and I got train accuracy ~0.7 but validation accuracy only ~0.5. 

But previous I run the same code on the earlier version of Keras and got pretty good accuracy on the validation data set. Then I also double checked accuracy of the model on the same training data set using the “evaluate_generator” function and I got loss and accuracy like: [0.7520096931622658, 0.5263203440548537]

So I guess that there are some bugs in the Keras 2.1.5.

**Comment by shinstra** [comment]
could be related to #4875
